src/lazy/binary/raw/v1_1/annotations_iterator.rs

```diff
@@ -1,23 +1,55 @@
 #![allow(non_camel_case_types)]
-use crate::lazy::binary::raw::v1_1::immutable_buffer::ImmutableBuffer;
-use crate::{IonResult, RawSymbolRef};
+use crate::lazy::binary::raw::v1_1::immutable_buffer::{AnnotationsEncoding, ImmutableBuffer};
+use crate::lazy::encoder::binary::v1_1::flex_sym::FlexSymValue;
+use crate::{IonResult, RawSymbolRef, SymbolId};
 
 /// Iterates over a slice of bytes, lazily reading them as a sequence of FlexUInt- or
 /// FlexSym-encoded symbol IDs.
 pub struct RawBinaryAnnotationsIterator_1_1<'a> {
     buffer: ImmutableBuffer<'a>,
+    encoding: AnnotationsEncoding,
 }
 
 impl<'a> RawBinaryAnnotationsIterator_1_1<'a> {
-    pub(crate) fn new(buffer: ImmutableBuffer<'a>) -> RawBinaryAnnotationsIterator_1_1<'a> {
-        Self { buffer }
+    pub(crate) fn new(
+        buffer: ImmutableBuffer<'a>,
+        encoding: AnnotationsEncoding,
+    ) -> RawBinaryAnnotationsIterator_1_1<'a> {
+        Self { buffer, encoding }
     }
 }
 
 impl<'a> Iterator for RawBinaryAnnotationsIterator_1_1<'a> {
     type Item = IonResult<RawSymbolRef<'a>>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        todo!()
+        if self.buffer.is_empty() {
+            return None;
+        }
+        use AnnotationsEncoding::*;
+        let (raw_symbol, remaining_input) = match self.encoding {
+            SymbolAddress => match self.buffer.read_flex_uint() {
+                Ok((flex_uint, remaining_input)) => (
+                    RawSymbolRef::SymbolId(flex_uint.value() as SymbolId),
+                    remaining_input,
+                ),
+                Err(error) => return Some(Err(error)),
+            },
+            FlexSym => {
+                let (flex_sym, remaining_input) = match self.buffer.read_flex_sym() {
+                    Ok((flex_sym, remaining_input)) => (flex_sym, remaining_input),
+                    Err(error) => return Some(Err(error)),
+                };
+                let raw_symbol = match flex_sym.value() {
+                    FlexSymValue::SymbolRef(raw_symbol) => raw_symbol,
+                    FlexSymValue::Opcode(_) => {
+                        todo!("FlexSym escapes in annotation sequences")
```

Comment:
Are `$0` and `''` going to be handled here or is that handled elsewhere so that it falls under the `SymbolRef` case?

Reply:
@glitch can comment on his intent, but I believe that's the idea.